src/Connection.php

```diff
@@ -159,7 +165,21 @@ private function receive($len = null)
     {
 
         if ($len) {
-            $line = fread($this->streamSocket, $len);
+            $chunkSize = $this->chunkSize;
+            $line = null;
+            $receivedBytes = 0;
+            while ($receivedBytes < $len) {
+                $bytesLeft = $len - $receivedBytes;
+                if ( $bytesLeft < 1500 ) {
+                    $chunkSize = $bytesLeft;
+                }
+
+                $line .= fread($this->streamSocket, $chunkSize);
+                $receivedBytes += $chunkSize;
+            }
+            if (strlen($line) > 2) {
+                $line = substr($line, 0, -2);
```

Comment:
You're probably right that I don't need to trim those, I'm using protobuf on the wire and had a corrupt message initially when I didn't trim those but that must have been caused by something else I was doing.